account_analytic_distribution_manual/models/account_analytic_distribution_manual.py

```diff
@@ -22,6 +32,34 @@ class AccountAnalyticDistributionManual(models.Model):
         ),
     ]
 
+    @api.depends("analytic_distribution")
+    def _compute_analytic_distribution_import(self):
+        aa_model = self.env["account.analytic.account"]
+        for item in self:
+            data = {}
+            distribution = item.analytic_distribution
+            for key in list(distribution.keys()):
+                aa_record = aa_model.browse(int(key))
+                data[aa_record.name] = distribution[key]
+            item.analytic_distribution_import = data
+
+    def _inverse_analytic_distribution_import(self):
+        """Convert the json to the appropriate value of analytic_distribution."""
+        aa_model = self.env["account.analytic.account"]
+        base_domain = [("company_id", "in", self.env.company.ids + [False])]
+        for item in self:
+            data = {}
+            new_distribution = item.analytic_distribution_import
+            for key in list(new_distribution.keys()):
+                domain = base_domain + [("name", "=", key)]
```

Comment:
Sorry, I see we can access the company on each `item`, so it’s better to use this field instead of the environment company.

```suggestion
        for item in self:
            base_domain = [("company_id", "in", item.company_id.ids + [False])]
            data = {}
            new_distribution = item.analytic_distribution_import
            for key in list(new_distribution.keys()):
                domain = base_domain + [("name", "=", key)]
```